File: QE_FST_Selenium_java/Activity11_waits1.java
package demos;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
//import org.openqa.selenium.interactions.Actions;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;
import java.time.Duration;

public class Activity11_waits1 {
    public static void main(String[] args) throws InterruptedException {
        WebDriver driver=new FirefoxDriver();
        //Actions actions=new Actions(driver);
        WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(10));
        driver.get("https://training-support.net/webelements/dynamic-controls");
        System.out.println(driver.getTitle());
        By checkbox = By.id("checkbox");
        WebElement ToggleButton = wait.until(ExpectedConditions.visibilityOfElementLocated(By.xpath("/html/body/div/main/div/div/div/div/div[2]/section[1]/button")));
        ToggleButton.click();
        //WebElement ToggleButton=driver.findElement(By.xpath(""));
        //actions.click(ToggleButton).perform();;
       wait.until(ExpectedConditions.invisibilityOfElementLocated(checkbox));
       System.out.println("Checkbox is removed (not visible).");
       ToggleButton.click();
        WebElement cb = wait.until(ExpectedConditions.visibilityOfElementLocated(checkbox));
        System.out.println("Checkbox is visible again.");
cb.click();
        driver.quit();


    }
}

File: QE_FST_Selenium_python/Activity11_waits1.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC

with webdriver.Firefox() as driver:
    driver.get("https://training-support.net/webelements/dynamic-controls")
    print(driver.title)
    wait = WebDriverWait(driver, 10)
    checkboxx=(By.ID,"checkbox")
    togglebtn=driver.find_element(By.XPATH,"//button[contains(text(),'Toggle')]")
    togglebtn.click()
    wait.until(EC.invisibility_of_element_located(checkboxx))
    print("Checkbox is disappeared")
    togglebtn.click()
    wait.until(EC.visibility_of_element_located(checkboxx))
    print("Checkbox is visible")
    driver.find_element(*checkboxx).click()
    if(driver.find_element(*checkboxx).is_selected):
        print("checkbox is selected")
